tauri-app/src/lib/texteditor/forkParse.ts

```diff
@@ -0,0 +1,59 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import { invoke } from '@tauri-apps/api';
+import { RainDocument, ErrorCode, type Problem } from "codemirror-rainlang";
+
+/**
+ * Parses a RainDocument with native parser with hardcoded entrypoints
+ * @param dotrain - The RainDocument instance
+ * @param forkUrl - frok url
+ * @param forkBlockNumber - fork block number
+ * @returns Resolves with empty array or with array of Problems if encountered an error
+ */
+export async function forkParseDotrain(dotrain: RainDocument, forkUrl: string, forkBlockNumber: number): Promise<Problem[]> {
+  let rainlang: string;
+  const frontMatter = dotrain.frontMatter;
+  try {
+    // set the hardcoded entrypoints
+    rainlang = await dotrain.compose(["calculate-io", "handle-io"]);
```

Comment:
I'd prefer to have the compose into the rust function in common.

Reply:
well I didnt do that to not trigger another ound of dotrain parsing, the RainDocument instance passed here is already parsed, so there is no need to reparse it again in rust
also I dont really get the need for waht you are suggesting, this is only for text editor, I dont know how it being dont on rust side helps with anything

Reply:
this is specifically for text editor consumption, so if you'd need a similar function but the compose happening in rust, feel free to make that

Reply:
here the parsed RainDocument comes from the text editor, as the callback

Reply:
also this is living in rust, all RainDocument instances live in rust, not js, as they just interfaced through wasm bindings for js, so compose() here is actually calling on a rust command for composing

Reply:
the issue with what you are suggesting is that there will be another round of parsing on the dotrain string, since passing the RainDocument instance from js to tauri and then to rust will result in loosing the RainDocument instance, and it needs to built again with the original string,
or if you pass the RainDocument instance itself it will serialized/deserialized which again will result in a new instance of RainDocument which is not performant and not suitable for editor

Reply:
> > well I didnt do that to not trigger another round of dotrain parsing, the RainDocument instance passed here is already parsed, so there is no need to reparse it again in rust
> 
> You are calling fork_parse, after the compose succeeds. I'm saying do the compose and throw the error from the rust implementation in common. I'm going to use the same rust implementation for the cli, so I don't want duplicate code paths ideally.

if you want a command that composes the dotrain in rust side, I would say you need to make that one specifically, as it would be doing a different thing

Reply:
ah I see makes sense. I'll merge as-is and see if I can unify the code paths in a refactor.